client/src/components/App.js

```diff
@@ -35,6 +43,56 @@ class App extends Component {
     this.setState({projects});
   }
 
+  parseProjectForm(form_data) {
+    for (let key in form_data) {
+      if (form_data[key] === '') {
+        // Actually delete so that we don't send empty strings or undefined to the database.
+        delete form_data[key];
+      }
+    }
+    if (!form_data.title || !form_data.link || !form_data.email) {
+      // TODO: show error here
+      return;
+    }
+    return form_data;
+  }
+
+  async submitNewProject(form_data) {
+    let data = this.parseProjectForm(form_data);
```

Comment:
It looks like we might return an empty object from parseProjectForm, but we don't handle it at all here. Should we skip the POST in that case?